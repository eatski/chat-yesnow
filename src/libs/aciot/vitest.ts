import { AxiosInstance } from "axios";
import { AciotCore, CacheMode } from "./core";
import { afterAll, beforeAll } from "vitest";
import path from "path";
import filenamify from 'filenamify';

export const applyTestHooks = (
	axios: AxiosInstance,
	config: {
		mode: CacheMode;
	},
) => {
	let aciot: AciotCore | null;
	beforeAll(({ name, file }) => {
		if (!file?.name) {
			throw new Error("suite.file.name is not defined");
		}
		const parsedPath = path.parse(file.name);
		const filename = filenamify(name, { replacement: "_" });
		aciot = new AciotCore({
			cacheBasePath: path.resolve(
				process.cwd(),
				parsedPath.dir,
				"__data__",
				parsedPath.name + parsedPath.ext,
				filename
			),
		});
		aciot.applyInterceptors(axios, config.mode);
	});
	afterAll(async ({ name, tasks }) => {
		if (tasks.every((t) => t.result?.state === "pass")) {
			console.info(name, "Clearing unused cache...");
			await aciot?.clearUnusedCache();
		}
	});
};
