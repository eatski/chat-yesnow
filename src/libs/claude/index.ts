import {
	Message,
	type MessageCreateParamsNonStreaming,
} from "@anthropic-ai/sdk/resources";

if (!process.env["ANTHROPIC_API_KEY"]) {
	throw new Error("ANTHROPIC_API_KEY is not defined");
}

const apiKey = process.env["ANTHROPIC_API_KEY"];

export const createMessage = (config: MessageCreateParamsNonStreaming) => {
	return fetch("https://api.anthropic.com/v1/messages", {
		headers: {
			"Content-Type": "application/json",
			"x-api-key": apiKey,
			"anthropic-version": "2023-06-01",
		},
		method: "POST",
		body: JSON.stringify(config),
	}).then((res) => res.json() as Promise<Message>);
};
