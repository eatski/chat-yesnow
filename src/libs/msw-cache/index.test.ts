import { describe, expect, test } from "vitest";
import { anthropic } from "../claude";
import { applyTestHooks } from "./vitest";

describe("initCacheMswServer", () => {
	applyTestHooks();
	test("anthropic", async () => {
		const result = await anthropic.messages.create({
			model: "claude-3-opus-20240229",
			messages: [
				{
					role: "user",
					content: "こんにちわ",
				},
			],
			max_tokens: 1000,
			stream: false,
		});
		expect(result.content).toMatchSnapshot();
	});
});
