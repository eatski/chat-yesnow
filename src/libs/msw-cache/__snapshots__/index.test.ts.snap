// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`initCacheMswServer > anthropic 1`] = `
[
  {
    "text": "Hello! How can I",
    "type": "text",
  },
]
`;

exports[`initCacheMswServer > embeddings 1`] = `3`;

exports[`initCacheMswServer > fetch 1`] = `
{
  "abbreviation": "WAT",
  "client_ip": "240b:10:1f20:2c00:841:54a1:c6c6:6dc6",
  "datetime": "2024-03-17T11:27:39.904055+01:00",
  "day_of_week": 0,
  "day_of_year": 77,
  "dst": false,
  "dst_from": null,
  "dst_offset": 0,
  "dst_until": null,
  "raw_offset": 3600,
  "timezone": "Africa/Lagos",
  "unixtime": 1710671259,
  "utc_datetime": "2024-03-17T10:27:39.904055+00:00",
  "utc_offset": "+01:00",
  "week_number": 11,
}
`;

exports[`initCacheMswServer > openai 1`] = `
[
  {
    "finish_reason": "length",
    "index": 0,
    "logprobs": null,
    "message": {
      "content": "I am a virtual assistant",
      "role": "assistant",
    },
  },
]
`;
